server/tabby_server/api/books.py

```diff
@@ -10,38 +22,156 @@ def books_scan_cover():
     """Receives an image and returns a list of possible books that the image
     could represent.
 
-    Expected fields in JSON:
-    - `"image"`: Base64 data representing the image.
+    The body of the request should be binary data (JPG or PNG) reprsenting
+    the image.
     """
 
-    if not request.is_json:
-        return {
-            "message": "Content type must be JSON."
-        }, HTTPStatus.BAD_REQUEST
-    data = request.get_json()
+    logging.info("scanning image")
 
-    image = data.get("image")
-    if not image:
+    # Try scan image
+    try:
+        img = Image.open(BytesIO(request.data))
+    except PIL.UnidentifiedImageError:
         return {
-            "message": "Must specify 'image' as a non-empty string in body."
+            "message": "Couldn't read an image from the given body."
         }, HTTPStatus.BAD_REQUEST
 
-    return {"results": []}, HTTPStatus.OK
+    img = img.convert("RGB")
+
+    logging.info("creating matrix")
+
+    # ai-gen start (ChatGPT-4o, 2)
+    img_mat = np.array(img)
+    img_mat = cv.cvtColor(img_mat, cv.COLOR_RGB2BGR)
+    # ai-gen end
+
+    logging.info("scanning cover")
+
+    books = scan_cover(img_mat)
+    logging.info("filtering book results")
+
+    # Filter out books without ISBNs
+    books = [b for b in books if b.isbn]
+
+    # Wrap it up in another dictionary and send!
+    result = _get_result_dict(books)
+    return result, HTTPStatus.OK
+
+
+def scan_cover(image: MatLike) -> list[google_books.Book]:
+    """Takes in an image of a cover and returns a list of results.
+
+    Args:
+        image: Image to scan.
```

Comment:
Just for clarity, I think you should add what datatype the function excepts to receive. I see that it says MatLike so I assume an image matrix so explicitly saying that would be helpful for other people using the function

Reply:
Good point, since we are dealing with multiple datatypes (`PIL.Image` and `MatLike`) that could represent an image.